atom/browser/osr/osr_host_display_client.cc

```diff
@@ -46,31 +49,11 @@ void LayeredWindowUpdater::OnAllocatedSharedMemory(
     return;
 
 #if defined(WIN32)
-  base::SharedMemoryHandle shm_handle;
-  size_t required_bytes;
-  MojoResult unwrap_result = mojo::UnwrapSharedMemoryHandle(
-      std::move(scoped_buffer_handle), &shm_handle, &required_bytes, nullptr);
-  if (unwrap_result != MOJO_RESULT_OK)
-    return;
-
-  base::SharedMemory shm(shm_handle, false);
-  if (!shm.Map(required_bytes)) {
-    DLOG(ERROR) << "Failed to map " << required_bytes << " bytes";
-    return;
-  }
-
   canvas_ = skia::CreatePlatformCanvasWithSharedSection(
-      pixel_size.width(), pixel_size.height(), false, shm.handle().GetHandle(),
-      skia::CRASH_ON_FAILURE);
+      pixel_size.width(), pixel_size.height(), false,
+      region.GetPlatformHandle(), skia::CRASH_ON_FAILURE);
 #else
-  auto shm =
-      mojo::UnwrapWritableSharedMemoryRegion(std::move(scoped_buffer_handle));
-  if (!shm.IsValid()) {
-    DLOG(ERROR) << "Failed to unwrap shared memory region";
-    return;
-  }
-
-  shm_mapping_ = shm.Map();
+  shm_mapping_ = region.Map();
```

Comment:
While `shm_mapping_` is retained by the class, the readable `region` is only available for the scope of this function, do we want to retain it as well ?

```
shm_region_ = std::move(region);
```

I am not sure whats the reason behind retaining `shm_mapping_` , /cc @brenca who wrote the initial implementation.

Reply:
Me either, this code is a mystery to me. The tests pass though? 😬

Reply:
linux seemed to have related failure https://circleci.com/gh/electron/electron/229118?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link , rerunning it now.

Reply:
The linux ones are failing consistently.

Reply:
ah... you're right. okay, i guess we'll have to hold off on merging then.